homeassistant/components/sensor/crimereports.py

```diff
@@ -0,0 +1,134 @@
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.components.zone import (
+    ATTR_RADIUS, ENTITY_ID_FORMAT as ZONE_ENTITY_ID_FORMAT)
+from homeassistant.const import (
+    CONF_INCLUDE, CONF_EXCLUDE, CONF_ZONE,
+    ATTR_ATTRIBUTION, ATTR_LATITUDE, ATTR_LONGITUDE, ATTR_FRIENDLY_NAME,
+    LENGTH_KILOMETERS, LENGTH_METERS)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import slugify, Throttle
+from homeassistant.util.distance import convert
+from homeassistant.util.dt import now
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['crimereports==1.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['zone']
+DOMAIN = 'crimereports'
+CONF_UPDATE_INTERVAL = 'update_interval'
+EVENT_INCIDENT = '{}_incident'.format(DOMAIN)
+NAME_FORMAT = '{} Incidents'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    CONF_ZONE: cv.string,
+    vol.Optional(CONF_INCLUDE): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_EXCLUDE): vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_UPDATE_INTERVAL, default=timedelta(seconds=1800)): (
+        vol.All(cv.time_period, cv.positive_timedelta)),
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Crime Reports platform."""
+    zone_id = ZONE_ENTITY_ID_FORMAT.format(config.get(CONF_ZONE))
+    zone_state = hass.states.get(zone_id)
+    if not zone_state:
+        _LOGGER.error("could not find specified zone: %s", zone_id)
+        return
+    latitude = zone_state.attributes.get(ATTR_LATITUDE)
+    longitude = zone_state.attributes.get(ATTR_LONGITUDE)
+    radius = zone_state.attributes.get(ATTR_RADIUS)
+    zone_friendly_name = zone_state.attributes.get(ATTR_FRIENDLY_NAME)
+    name = NAME_FORMAT.format(zone_friendly_name)
+    add_devices([CrimeReportsSensor(hass, name, latitude, longitude, radius,
+                                    config.get(CONF_INCLUDE),
+                                    config.get(CONF_EXCLUDE),
+                                    config.get(CONF_UPDATE_INTERVAL))])
+
+
+class CrimeReportsSensor(Entity):
+    """Crime Reports Sensor."""
+
+    def __init__(self, hass, name, latitude, longitude, radius,
+                 include, exclude, interval):
+        """Initialize the sensor."""
+        import crimereports
+        self._hass = hass
+        self._name = name
+        self._include = include
+        self._exclude = exclude
+        radius_kilometers = convert(radius, LENGTH_METERS, LENGTH_KILOMETERS)
+        self._crimereports = crimereports.CrimeReports((latitude, longitude),
+                                                       radius_kilometers)
+        self._attributes = None
+        self._state = None
+        self._previous_incidents = set()
+        self.update = Throttle(interval)(self._update)
+        self.update()
```

Comment:
Use `SCAN_INTERVAL = timedelta()` on top of your script and remove this update here. You can call `add_devices(..., True)` for make a update bevor he add this to statemachine.